models/comments.model.js

```diff
@@ -22,5 +13,17 @@ async function selectCommentsByArticleId(articleId) {
     return commentsResult.rows   
 }
 
+async function createComment(articleId, author, body) {
+    await selectArticleById(articleId)
+
+    await selectUserByUsername(author)
```

Comment:
article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value. 

This means you don't need to check separately (comment these checks out and run the code to see what I mean - you don't currently have PSQL error handlers from what I can see, but you can console.log the error and see how you may need to handle it).

Reply:
Is it better to send an invalid INSERT and then deal with the conequences than to validate the values first? It would potentially make just one database visit rather than 3 but it feels wrong to not check stuff we can validate against a DB table.